source/fundamentals/connection.txt

```diff
@@ -4,16 +4,185 @@ Connection Guide
+
+   If your deployment is on MongoDB Atlas, see the
+   :atlas:`Atlas driver connection guide </driver-connection>`
+   and select Go from the language dropdown to retrieve your connection string.
+
+The next part of the connection string contains your username and, if
+you are using password-based authentication, your password. Replace the value of
+``user`` with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this section of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the example above, we use ``sample-hostname``
+as the hostname and ``27017``, the default port, as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string specifies connection and authentication
+options. In the example, we set two connection options:
+``poolSize=20`` and ``writeConcern=majority``. For more information on connection
+options, read the :ref:`connection-options` section of this guide.
+
+The code below shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.go
+   :language: go
+
+Connect to ``localhost``
+++++++++++++++++++++++++
+
+.. include:: /includes/localhost-connection.rst
+
+------------------------
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname and port numbers
+of each instance, separated by commas, and the replica set name as the value
+of the ``replicaSet`` parameter in the connection string. In the following
+example, the replica set name is ``myRS``.
+
+.. code-block:: none
+
+   mongodb://host1:27017,host2:27017,host3:27017/?replicaSet=myRS
+
+When connecting to a replica set, the driver takes the following actions by default:
+
+- Discovers all replica set members when given the address of any one member.
+- Dispatches operations to the appropriate member, such as a directive
+  to write against the **primary**.
+
+.. tip::
+
+   You only need to specify one host to connect to a replica set. However, to ensure connectivity if the specified host
+   is unavailable, you should provide the full list of hosts.
+
+Direct Connection
++++++++++++++++++
+
+To force operations on the host designated in the connection URI, specify the ``directConnection`` option. Direct
+connections:
+
+- Don't support SRV strings.
+- Fail on writes when the specified host is not the **primary**.
+- Require you to :manual:`enable secondary reads </reference/method/rs.secondaryOk/>`
+  when the specified host isn't the **primary**.
+
+.. _connection-options:
+
+------------------
+Connection Options
```

Comment:
This section looks good! But, I dug around a bit but couldn't find an exact link to verify the table in this section. Can you provide the link you used  so I can take a look in the next round of reviews?

Reply:
The ones in the table are the ones from [this link](https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.7.0/mongo/options#ClientOptions) that also say you can set them in the URI options. Added a tip box to the page with the same link.

Reply:
the order is currently random, let me know if that should be changed.

Reply:
I think the ordering is fine, but for the tip box, I think it should be a statement at the end of the table like it is in the [Java docs](https://docs.mongodb.com/drivers/java/sync/current/fundamentals/connection/#connection-options) and [Node docs](https://docs.mongodb.com/drivers/node/current/fundamentals/connection/#connection-options)